### DISCUSSION TITLE
Is support for Tail Call Optimization planned?

### DISCUSSION THREAD

**Discussion by pierre-rouleau** [vlang/v#9428] [General]
Is there a plan to support Tail Call Optimization V to allow efficient use of recursive functions?  I did not see any mention of TCO in the docs.

**Comment by JalonSolov** [comment]
That is a long way off.  The default C backend should do it for you in `-prod` compiles.

**Comment by medvednikov** [comment]
It's already there as long as you use `-prod`.

**Reply by pierre-rouleau** [reply]
Great.  Is there any advantage to not use -prod for other reasons?

**Reply by Gladear** [reply]
Compilation without `-prod` is much faster, it should be reserved for production build (as the name suggests) where performances are more important than build time 🙂

**Reply by pierre-rouleau** [reply]
Thanks for the answers!

Several questions come to mind:

* Is TCO supported with -prod for all supported target C compilers?  
  * For instance, does that mean that -O3 is used when GCC is used? 

* On the topic of -prod:
   * aside from optimizations, verifications of unused variables and embedding, what else differ?  Is there a list somewhere?
   * is debugging with -prod compiled code supported?
   * when creating and run-in unit test code, would you use -prod ? 
     * while it's good to get a result quickly, you'd probably want to ensure that your test reflect what's going to run at production time.  What is the strategy mostly used here?

**Reply by medvednikov** [reply]
Yes, O3 is used with -prod

Nothing else is different.

Yes, you can do -cg -prod

-prod tests is a good idea, but it slows it down a lot

**Reply by pierre-rouleau** [reply]
``"-prod tests is a good idea, but it slows it down a lot"``

I assume that you mean that the compilation will be slower (affecting the overall test time), but I'd assume that the execution would not suffer slowdown.  Is my assumption correct?
